app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupingMakingFragment.java

```diff
@@ -70,27 +79,62 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
         settings = new GroupingSettings(
+                listInfo.getNumNames(),
                 getResources().getInteger(R.integer.default_number_of_names_per_group),
                 getResources().getInteger(R.integer.default_number_of_groups));
         settingsDialog = new GroupingSettingsDialog(getActivity(), settings);
     }
 
+    @Override
+    public void onNameAdded(String name, int amount, String listName) {
+        listInfo = new DataSource(getContext()).getListInfo(listName);
```

Comment:
Instead of creating a new DataSource every time, I think it's worth just having a member variable. It will save calls to getContext() as well.